chia/cmds/coin_funcs.py

```diff
@@ -2,79 +2,88 @@
-from chia.rpc.wallet_rpc_client import WalletRpcClient
 from chia.types.blockchain_format.coin import Coin
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.coin_record import CoinRecord
 from chia.util.bech32m import decode_puzzle_hash, encode_puzzle_hash
+from chia.util.config import selected_network_address_prefix
 from chia.util.ints import uint64, uint128
 from chia.wallet.transaction_record import TransactionRecord
 from chia.wallet.util.wallet_types import WalletType
 
 
-async def async_list(args: Dict[str, Any], wallet_client: WalletRpcClient, fingerprint: int) -> None:
-    wallet_id: int = args["id"]
-    min_coin_amount = Decimal(args["min_coin_amount"])
-    max_coin_amount = Decimal(args["max_coin_amount"])
-    excluded_coin_ids = args["excluded_coin_ids"]
-    excluded_amounts = args["excluded_amounts"]
-    addr_prefix = args["addr_prefix"]
-    show_unconfirmed = args["show_unconfirmed"]
-    paginate = args["paginate"]
-    if paginate is None:
-        paginate = sys.stdout.isatty()
-    try:
-        wallet_type = await get_wallet_type(wallet_id=wallet_id, wallet_client=wallet_client)
-        mojo_per_unit = get_mojo_per_unit(wallet_type)
-    except LookupError:
-        print(f"Wallet id: {wallet_id} not found.")
-        return
-    if not await wallet_client.get_synced():
-        print("Wallet not synced. Please wait.")
-        return
-    final_min_coin_amount: uint64 = uint64(int(min_coin_amount * mojo_per_unit))
-    final_max_coin_amount: uint64 = uint64(int(max_coin_amount * mojo_per_unit))
-    final_excluded_amounts: List[uint64] = [uint64(int(Decimal(amount) * mojo_per_unit)) for amount in excluded_amounts]
-    conf_coins, unconfirmed_removals, unconfirmed_additions = await wallet_client.get_spendable_coins(
-        wallet_id=wallet_id,
-        max_coin_amount=final_max_coin_amount,
-        min_coin_amount=final_min_coin_amount,
-        excluded_amounts=final_excluded_amounts,
-        excluded_coin_ids=excluded_coin_ids,
-    )
-    print(f"There are a total of {len(conf_coins) + len(unconfirmed_additions)} coins in wallet {wallet_id}.")
-    print(f"{len(conf_coins)} confirmed coins.")
-    print(f"{len(unconfirmed_additions)} unconfirmed additions.")
-    print(f"{len(unconfirmed_removals)} unconfirmed removals.")
-    print("Confirmed coins:")
-    print_coins(
-        "\tAddress: {} Amount: {}, Confirmed in block: {}\n",
-        [(cr.coin, str(cr.confirmed_block_index)) for cr in conf_coins],
-        mojo_per_unit,
-        addr_prefix,
-        paginate,
-    )
-    if show_unconfirmed:
-        print("\nUnconfirmed Removals:")
-        print_coins(
-            "\tPrevious Address: {} Amount: {}, Confirmed in block: {}\n",
-            [(cr.coin, str(cr.confirmed_block_index)) for cr in unconfirmed_removals],
-            mojo_per_unit,
-            addr_prefix,
-            paginate,
+async def async_list(
+    wallet_rpc_port: Optional[int],
```

Comment:
when you have this many parameters, some of them having the same type, I think it might be appropriate to add `*,` at the beginning of the parameter list, to require callers to name the arguments.

I think this would make sense for the other functions too

Reply:
Just finished changing all the long ones.